### PR TITLE
To Do List Project: Interactivity

### PR DESCRIPTION
**Pull request by JoyTaribagshaw**
**Project requirements**
>

1. Contain methods related to the status updates (completed: true / false).
2. Add event listener to the checkbox (change).
3. Update items object's value for completed key upon user actions.
4. Implement a function for the "Clear all completed" button (use filter() method).
5. Store the updated array of items in local storage, so the user gets the correct list of values after the page reloads (which means that any changes made to the list should be preserved).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,36 +1,150 @@
...
+      if (list.index === index) {
+        list.completed = !list.completed;
+      }
+      return list;
+    });
+  };
+
+  clearAllCompletedTasks = () => {
+    this.myToDoList = this.myToDoList.filter((list) => !list.completed);
+  };
```
Comment (Munsa1) [inline src/index.js]: - [ ] Its a requirement that we should create a new JavaScript file to hold the functionality for this milestone. Kindly do so.Check the screenshot of the requirements below
image


- [ ] Good job implementing the task object's values for `completed` and adding a function that deletes all completed tasks. However, we have some issues we need to fix. The user is unable to add and delete tasks to the app, check the screencast below. Kindly fix this issue.

https://github.com/JoyTaribagshaw/To-Do-List-Project/assets/81932478/5cd45ccd-aebd-4fa5-a14d-197c84486a22

File: src/index.js [hunk 2]
```diff
@@ -1,36 +1,150 @@
...
+    });
+  };
+
+  clearCompletedTasksFromUI = () => {
+    clearBtn.addEventListener('click', () => {
+      this.clearAllCompletedTasks();
+      this.setLocalStorage();
+      this.populateToDoList();
+    });
+  }
```
Comment (Munsa1) [inline src/index.js]: - [ ] Kindly ensure that once the error causing tasks to not be added is resolved, the array of items stored in local storage should be updated after each user action, so the user gets the correct list values after the page reloads.

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @JoyTaribagshaw, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Added A Filter Function For Deleting Completed Tasks :heavy_check_mark: 
Added A Function Which Changes Tasks To Completed :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (JoyTaribagshaw) [reply src/index.js]: Thank you so much for your review @Munsa1 , I have resolved the local storage error.

File: src/index.js (hunk 1 above)
Comment (JoyTaribagshaw) [reply src/index.js]: Thank you so much for your review @Munsa1 , I have resolved the add and delete tasks functionality.

**Review by Davidosky007 — APPROVED** [review]
Hi @JoyTaribagshaw 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Webpack setup properly 👍 
* Good looking readme 👍 
* ES6 syntax used properly 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**